std/random.d

```diff
@@ -1183,6 +1183,135 @@ A singleton instance of the default random number generator
 }
 
 /**
+A random bool generator that has the underlying random integral
+generation batched. The underlying random number generator can be
+passed as a pointer so the RNG will not get copied internally.
+ */
+struct RandomBool(Rng = Random)
+    if (isUniformRNG!Rng || isUniformRNG!(typeof(*Rng.init)))
+{
+    private Rng parent;
+    private uint mask;
+    private uint accumulator;
+
+    /// Always `false`
+    enum bool empty = false;
+
+    /**
+      If Rng is not a pointer, we want to pass it by ref
```

Comment:
This should not be part of the user-visible documentation because it's a mere optimization. Just describe what it does and let the user infer the optimization from the signature if they are interested.